pkg/reconciler/revision/revision.go

```diff
@@ -97,70 +97,35 @@ func (c *Reconciler) reconcileDigest(ctx context.Context, rev *v1.Revision) erro
 	}
 
 	var digestGrp errgroup.Group
-	type digestData struct {
-		digestValue        string
-		containerName      string
-		isServingContainer bool
-		image              string
-		digestError        error
-	}
-
-	digests := make(chan digestData, len(rev.Spec.Containers))
-	containerIndexMap := map[string]int{}
+	rev.Status.ContainerStatuses = make([]v1.ContainerStatuses, len(rev.Spec.Containers))
 	for i, container := range rev.Spec.Containers {
-		containerIndexMap[container.Name] = i
 		container := container // Standard Go concurrency pattern.
+		i := i
 		digestGrp.Go(func() error {
 			digest, err := c.resolver.Resolve(container.Image,
 				opt, cfgs.Deployment.RegistriesSkippingTagResolving)
 			if err != nil {
 				err = fmt.Errorf("failed to resolve image to digest: %w", err)
-				digests <- digestData{
-					image:       container.Image,
-					digestError: err,
-				}
+				rev.Status.MarkContainerHealthyFalse(v1.ReasonContainerMissing,
```

Comment:
think this is racey unfortunately cos we can fail in parallel, probably need to do this after the loop

Reply:
That was my suggestion to @savitaashture  in the other PR to just use `sync.Once`, though won't help with return.

Reply:
cant we just do something like

````
err := digestGrp.Wait(); 
if err != nil {
 	rev.Status.MarkContainerHealthyFalse(v1.ReasonContainerMissing, err)
}
return err
````
?

Reply:
now that we're filling everything in line, probably no reason why not.

Reply:
We can't do this because the Mark function takes a containerName too. We'd need a special error type to do that.

Reply:
Yeah that (special error type) is what I was suggesting in https://github.com/knative/serving/pull/7866#discussion_r421011242. Though sync.Once would work too if necc.